### PR TITLE
Add elphmod example

### PR DESCRIPTION
**Pull request by janberges**
This example shows how to interface i-PI with [elphmod](https://github.com/janberges/elphmod) as done in [SciPost Phys. **16**, 046 (2024)](https://doi.org/10.21468/SciPostPhys.16.2.046).

### CODE REVIEW

**Comment by litman90** [comment]
Hi Mariana, is this ready for review? If I get your confirmation by then I will review it on Monday

**Comment by mahrossi** [comment]
yes! ready to review

File: examples/clients/elphmod/README.md
Comment (litman90) [inline examples/clients/elphmod/README.md:37]: same as above

File: examples/clients/elphmod/run.py
Comment (litman90) [inline examples/clients/elphmod/run.py:8]: I would leave this as an **alternative** option  adding clearly that it will only work  when i-PI has been installed through pip

File: examples/clients/elphmod/driver.py
Comment (litman90) [inline examples/clients/elphmod/driver.py:4]: Please, explain the main keywords of elphmod used here or add a doc string with the appropriate links to the elphmod documentation

**Comment by janberges** [comment]
Many thanks for considering this example!

I am happy to use `i-pi-driver-py`. As far as I understand, this requires adding a module to `drivers/py/pes/`, which is what kept me from using it so far (to have the whole example in one place). Since `elphmod.md.Driver` has many parameters, it might be the best to have that module load a preinitialized driver from a file whose name is the only argument (see my last commit). Would that be fine with you?

File: examples/clients/elphmod/README.md
Comment (janberges) [reply examples/clients/elphmod/README.md:12]: I agree. I use `i-pi-driver-py` now, see my comment below. (As a consequence, the example will only work with the next PyPI release of i-PI of course.)

File: examples/clients/elphmod/run.py
Comment (janberges) [reply examples/clients/elphmod/run.py]: For now, I have fully removed this option to keep it simple. If you think it is useful, I can bring back the removed script `run.py` and add an alternative phony target to the makefile that uses it.

File: examples/clients/elphmod/driver.py
Comment (janberges) [reply examples/clients/elphmod/driver.py:6]: Thanks for the hint! Is the minimal docstring with the link to the documentation of the relevant module enough? (I have chosen the Markdown format because it is used in some other docstrings of the repository.)

Comment (mahrossi) [reply examples/clients/elphmod/driver.py:6]: I would prefer to leave as Jan has it now, because describing all elphmod keywords here would be a repetition of work (that has been nicely done elsewhere).

**Comment by mahrossi** [comment]
Hey @janberges thanks for addressing all points, real great. So, one thing with the current pre-loading is that it becomes less transparent (at least to me), what would a user need to do if they wanted to use elphmod with their own (new) setup. I think this is a good compromise overall, but honestly, it would also help to actually have a more transparent setup. Before, it was more transparent. Is there a way to at least describe in the README what a user would have to do instead?

File: examples/clients/elphmod/driver.py
Comment (litman90) [reply examples/clients/elphmod/driver.py:6]: yes, that is enough thanks

**Review by litman90 — COMMENTED** [review]
Please import 'sys' in drivers/py/pes/elphmod.py 
After that, we will be ready to merge. Thanks

File: examples/clients/elphmod/README.md
Comment (janberges) [reply examples/clients/elphmod/README.md]: The numbered instructions are now only for the `i-pi-driver-py` way of doing this. Anything else is discussed at the end.

File: examples/clients/elphmod/run.py
Comment (janberges) [reply examples/clients/elphmod/run.py:8]: OK, it is probably better to leave this as an alternative option, because it shows that also `elphmod.md.Driver` on its own (without the wrapper in `drivers/py/pes/elphmod.py`) works like the drivers defined in `drivers/py/pes/`. It can be tried using the new phony target `run2`, which is not used by default.

**Comment by janberges** [comment]
Thanks! (It seems I only ran `make pretty`, sorry.) To hopefully make the reasons behind creating an intermediate driver file more transparent, I have added a comment at the end of the README. Is it now clearer that one can define a custom setup as in the script `driver.py`?

**Review by litman90 — APPROVED** [review]

**Comment by litman90** [comment]
Thanks Jan for your contribution.
From my side, it is ready to merge (provided all the tests pass).
@mahrossi, I am not sure if you wanted to ask Jan something more before merging.
Feel free to merge if you are satisfied.

**Comment by mahrossi** [comment]
I did ask in my comment above.

**Pull request merged**
